CONTRIBUTING.md

```diff
@@ -102,6 +102,7 @@ Pull requests will be merged based on the following criteria:
 - Does not have the `wip` label.
 - All status checks have succeeded.
 - If the person who opened the pull request is a core maintainer, then only that person is expected to merge once it has the necessary LGTMs/reviews. Another maintainer can merge the pull request at their discretion if they feel the pull request must be merged urgently.
+- Core maintainers are allowed to use the `squash and merge` option only for their own pull requests.
```

Comment:
If the reason we didn't want to squash and merge was to prevent the one merging from accidentally removing the DCO sign-off for the new commit on `main`, doesn't automating that with the bot eliminate that concern? What's the issue with squashing and merging for non-maintainers' contributions?

Reply:
This PR does not attempt to change the existing process, rather just documents it :-)

When we do allow the squash option for external contributors (not the case currently), this should be removed.